pkg/ratelimit/option.go

```diff
@@ -16,39 +16,118 @@ package ratelimit
 
 import "golang.org/x/time/rate"
 
+// UpdateStatus is flags for updating limiter config.
+type UpdateStatus uint32
+
+// Flags for limiter.
+const (
+	// QPSNoChange shows that limiter's config isn't changed.
+	QPSNoChange UpdateStatus = 1 << iota
+	// QPSChanged shows that limiter's config is changed and not deleted.
+	QPSChanged
+	// QPSDeleted shows that limiter's config is deleted.
+	QPSDeleted
+	// ConcurrencyNoChange shows that limiter's config isn't changed.
+	ConcurrencyNoChange
+	// ConcurrencyChanged shows that limiter's config is changed and not deleted.
+	ConcurrencyChanged
+	// ConcurrencyDeleted shows that limiter's config is deleted.
+	ConcurrencyDeleted
+	// InAllowList shows that limiter's config isn't changed because it is in in allow list.
+	InAllowList
+)
+
 // Option is used to create a limiter with the optional settings.
 // these setting is used to add a kind of limiter for a service
-type Option func(string, *Limiter)
+type Option func(string, *Limiter) UpdateStatus
 
 // AddLabelAllowList adds a label into allow list.
 // It means the given label will not be limited
 func AddLabelAllowList() Option {
-	return func(label string, l *Limiter) {
+	return func(label string, l *Limiter) UpdateStatus {
 		l.labelAllowList[label] = struct{}{}
+		return 0
 	}
 }
 
+func updateConcurrencyConfig(l *Limiter, label string, limit uint64) UpdateStatus {
+	l.configMux.Lock()
+	defer l.configMux.Unlock()
+
+	cfg, ok := l.labelConfig[label]
+	if !ok {
+		cfg = &DimensionConfig{}
+		l.labelConfig[label] = cfg
+	}
+	if cfg.ConcurrencyLimit == limit {
+		return ConcurrencyNoChange
+	}
+	cfg.ConcurrencyLimit = limit
+	l.labelConfig[label] = cfg
+	if limit < 1 {
```

Comment:
Meanwhile, the function `cfg.ConcurrencyUnlimit()`  may be more friendly?

Reply:
update